src/nupic/encoders/ScalarEncoder.cpp

```diff
@@ -78,7 +78,10 @@ void ScalarEncoder::initialize(ScalarEncoderParameters &parameters)
   }
 
   // Determine resolution & size.
-  const Real64 extentWidth = args_.maximum - args_.minimum;
+  Real64  maxInclusive     = args_.maximum;
+  UInt64* maxInclusiveData = (UInt64*)&maxInclusive;
```

Comment:
please use (?)`reinterpret_cast<UInt64>()` if you want to treat the binary representation as UIN64?

Reply:
Ok, will fix.

Im also going to make the endpoint exclusive if its periodic